File: src/main/resources/static/scripts/services/DetailsService.js
/**
 * Created by daniel.baker on 3/1/2015.
 */
(function(){
    var details = function($http) {
        var getDebts = function(budgIdSeq) {
            return $http.post("../getDebts", budgIdSeq)
                .then(function(response) {
                    return response.data;
                });
        };
        var getExpenses = function(budgIdSeq) {
            return $http.post("../getExpenses", budgIdSeq)
                .then(function(response) {
                    return response.data;
                });
        };
        var saveDebt = function(debtObj) {
            return $http.post("../saveDebt", debtObj)
                .then(function(response) {
                    return response.data;
                });
        };
        var saveExpense = function(expenseObj) {
            return $http.post("../saveExpense", expenseObj)
                .then(function(response) {
                    return response.data;
                });
        };

        return {
            getDebts: getDebts,
            getExpenses: getExpenses,
            saveDebt: saveDebt,
            saveExpense: saveExpense
        };
    };
    var module = angular.module("moneybags");
    module.factory("details", details);
}());

File: src/main/java/com/moneybags/services/DetailsService.java
package com.moneybags.services;

import com.moneybags.domain.Debt;
import com.moneybags.domain.Expense;
import org.springframework.stereotype.Service;

import java.util.List;

/**
 * Users: daniel.baker
 * Date: 3/8/2015
 * Time: 9:49 PM
 */
@Service
public interface DetailsService {
    void saveExpense(Expense expense);
    void saveDebt(Debt debt);
    List<Expense> getExpenses(Long budgIdSeq);
    List<Debt> getDebts(Long budgIdSeq);
}
